pipeline.py

```diff
@@ -112,32 +91,32 @@ def pipeline(
-        train_save_samples: int = SAVE_SAMPLES,
-        train_max_batch_len: int = MAX_BATCH_LEN,
-        train_seed: int = SEED,
+        train_nproc_per_node: int = 2,  # FIXME: Not present in default config. Arbitrary value chosen to demonstrate multi-node multi-gpu capabilities. Needs proper reference architecture justification.
+        train_nnodes: int = 2,  # FIXME: Not present in default config. Arbitrary value chosen to demonstrate multi-node multi-gpu capabilities. Needs proper reference architecture justification.
+        train_num_epochs_phase_1: int = 7,  # https://github.com/instructlab/instructlab/blob/v0.21.2/tests/testdata/default_config.yaml#L364
+        train_num_epochs_phase_2: int = 10,  # https://github.com/instructlab/instructlab/blob/v0.21.2/tests/testdata/default_config.yaml#L377
+        train_effective_batch_size_phase_1: int = 128,  # https://github.com/instructlab/instructlab/blob/v0.21.2/tests/testdata/default_config.yaml#L357
+        train_effective_batch_size_phase_2: int = 3840,  # https://github.com/instructlab/instructlab/blob/v0.21.2/tests/testdata/default_config.yaml#L371
+        train_learning_rate_phase_1: float = 2e-05,  # https://github.com/instructlab/instructlab/blob/v0.21.2/tests/testdata/default_config.yaml#L360
+        train_learning_rate_phase_2: float = 6e-06,  # https://github.com/instructlab/instructlab/blob/v0.21.2/tests/testdata/default_config.yaml#L374
+        train_num_warmup_steps_phase_1: int = 1000,  # https://github.com/instructlab/training/blob/v0.6.1/src/instructlab/training/main_ds.py#L874
+        train_num_warmup_steps_phase_2: int = 1000,  # https://github.com/instructlab/training/blob/v0.6.1/src/instructlab/training/main_ds.py#L874
+        train_save_samples: int = 250000,  # https://github.com/instructlab/instructlab/blob/v0.21.2/tests/testdata/default_config.yaml#L393
+        train_max_batch_len: int = 5000,  # https://github.com/instructlab/instructlab/blob/v0.21.2/tests/testdata/default_config.yaml#L334
+        train_seed: int = 42,  # https://github.com/instructlab/training/blob/v0.6.1/src/instructlab/training/main_ds.py#L901
         # MT Bench
-        mt_bench_max_workers: str = MAX_WORKERS,
-        mt_bench_merge_system_user_message: bool = MERGE_SYSTEM_USER_MESSAGE,
+        mt_bench_max_workers: str = "auto",  # https://github.com/instructlab/instructlab/blob/v0.21.2/tests/testdata/default_config.yaml#L74
+        mt_bench_merge_system_user_message: bool = False,  # https://github.com/instructlab/instructlab/blob/v0.21.2/src/instructlab/model/evaluate.py#L474
         # Final evaluation
-        final_eval_max_workers: str = MAX_WORKERS,
-        final_eval_few_shots: int = FEW_SHOTS,
-        final_eval_batch_size: str = BATCH_SIZE,
-        final_eval_merge_system_user_message: bool = MERGE_SYSTEM_USER_MESSAGE,
+        final_eval_max_workers: str = "auto",  # FIXME: Not present in default config. Need justification?
```

Comment:
https://github.com/instructlab/instructlab/blob/v0.21.2/tests/testdata/default_config.yaml#L74 this is used for both mt_bench and mt_bench_branch (AKA final eval :p)

Reply:
Should we also merge it into a single param then? WDYT @MichaelClifford ? 🙂 

Right now we have 2 separate params:
- `final_eval_max_workers` - used in MT Bench Branch
- `mt_bench_max_workers` - used in MT Bench

Reply:
I think we separated this out in an earlier PR since we didn't want to assume that two different processes will always need the same a parameter. However, I do see that in the default config they do not differentiate. 🤔 I'm fine keeping it as 2 separate params, but would also not care if we make it one. So its up to you @tumido 😄

Reply:
I'll keep it as is. If RHOAI/RHEL AI sees the need to unify these they always can. 🙂